src/devices/machine/tsconfdma.h

```diff
@@ -0,0 +1,57 @@
+// license:BSD-3-Clause
+// copyright-holders:Andrei I. Holub
+/***************************************************************************
+
+    TS-Conf (ZX-Evolution) DMA
+
+***************************************************************************/
+
+#ifndef MAME_MACHINE_TSCONF_DMA_H
+#define MAME_MACHINE_TSCONF_DMA_H
+
+#pragma once
+
+DECLARE_DEVICE_TYPE(TSCONF_DMA, tsconfdma_device)
+
+class tsconfdma_device : public device_t
+{
+public:
+    tsconfdma_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	auto in_mreq_callback() { return m_in_mreq_cb.bind(); }
+	auto out_mreq_callback() { return m_out_mreq_cb.bind(); }
+    auto in_spireq_callback() { return m_in_mspi_cb.bind(); }
+	auto out_cram_callback() { return m_out_cram_cb.bind(); }
+
+    bool is_ready();
```

Comment:
If this is supposed to be used as a read line callback, it should return `int` (or use the `DECLARE_READ_LINE_MEMBER` helper macro).